Draft: Allow template syntax in event text

luelista opened the pull request.

raphaelm commented on doc/development/api/placeholder.rst:
```diff
@@ -1,19 +1,20 @@
 
-Writing an e-mail placeholder plugin
+Writing a template placeholder plugin
 ====================================
```
```suggestion
=====================================
```

raphaelm reviewed (commented):
>  ⁉️ Should I change all refs in pretix core to the new location/names (get_email_context -> get_placeholder_context, etc)? Would be quite a bit of diff "noise", and we need to keep it available in the old places anyway because of plugins

Yea, I was also thinking about the `register_mail_placeholders` signal. I think adding the new ones and keeping the old ones around as deprecated aliases would be the "clean" way to go, even if a bit noisy :(

⁉️ Caching of get_placeholder_context result? Necessary? Best way to do it?

I think the internal cache of `PlaceholderContext` that you added is a good compromise for now.

⁉️ Placeholder-Funktionalität in weiteren Textfeldern (z.B. End of presale text, Additional success message) hinzufügen?

Let's keep it simple for now and see where we need it before we add too much complexity we have issues with later. For example, every field we add placeholder support too will be less useful for API clients who do not know how to evaluate it.

luelista commented on src/pretix/base/models/orders.py:
```diff
@@ -1090,9 +1090,6 @@ def send_mail(self, subject: Union[str, LazyI18nString], template: Union[str, La
             return
 
-        for k, v in self.event.meta_data.items():
-            context['meta_' + k] = v
```
Yes, as far as I can tell, all usages of these send_mail methods pass in the context as generated by get_email_context, which in turn contains the meta_ placeholders already. 

I also tested it with some order mails and the meta_ placeholders still work without this code.

luelista commented:
* I renamed the `register_mail_placeholders`signal to `register_text_placeholders`, and kept the old name around marked as deprecated.

* get_email_context: I wouldn't rename that, as it's still only used for email. The new event text templating uses the PlaceholderContext.

* SimpleFunctionalMailTextPlaceholder is internally renamed to SimpleFunctionalTextPlaceholder, but orig. name is still used in plugins - we can rename these references over time

* I also updated [the docs](https://github.com/pretix/pretix/pull/3815/files#diff-e871690e1d3bdd4b2aa548f7030b9867e5aed760fff216cc82341ea4eb1adf3bR7) some more.
